src/lib/serviceRoutines/postUpdateContext.cpp

```diff
@@ -188,6 +188,19 @@ std::string postUpdateContext
 
     cleanPayload = xmlPayloadClean(out.c_str(), "<updateContextResponse>");
 
+    if ((cleanPayload == NULL) || (cleanPayload[0] == 0))
+    {
+      UpdateContextResponse ucrs;
+
+      // FIXME P1: What StatusCode should be used here?
```

Comment:
I don't think this should be a 500 Internal Error. In general, fails in the connection with a CPr should be resolved as if there wasn't any CPr registered for the given context element, i.e. 404 Not Context Element found, from the perspective of the client. Of course, from an "internal" perspective it's perfectly ok (as you do some lines below) logging it as warning message in the log.